docs(api): add more 2.14 bugfixes to versioning doc (#12212)
OpenTrons/opentrons-api: api/docs/v2/versioning.rst

Version 2.14
++++++++++++

This version introduces a new protocol runtime that offers more reliable run control
and builds a strong foundation for future Protocol API improvements.

> Several older parts of the Protocol API deprecated as part of this switchover.

Reviewer: ```suggestion
Several older parts of the Protocol API were deprecated as part of this switchover.
```

  - :py:meth:`.ProtocolContext.load_labware` now prefers loading user-provided labware definitions
    rather than built-in definitions if no explicit ``namespace`` is specified.

  - :py:meth:`.ProtocolContext.pause` will now properly wait until you resume the protocol before moving on.

>     In previous versions, the run will not pause until the first call to a different `ProtocolContext` method.

Reviewer: ```suggestion
    In previous versions, the run will not pause until the first call to a different ``ProtocolContext`` method.
```
silly RST.

  - Motion planning has been improved to avoid certain erroneous downward movements,
    especially when using :py:meth:`.InstrumentContext.aspirate`.

>   - :py:attr:`.Labware.tip_length` will raise a useful error if called on a non-tip-rack.

Reviewer: ```suggestion
  - :py:attr:`.Labware.tip_length` will raise a useful error if called on labware that is not a tip rack.
```
avoiding some wonky punctuation here
